packages/fuel-gauge/src/auth-testing.test.ts

```diff
@@ -0,0 +1,54 @@
+import type { Contract } from 'fuels';
+
+import { getSetupContract, getWallet } from './utils';
+
+const setupContract = getSetupContract('auth_testing_contract');
+
+let contractInstance: Contract;
+
+beforeAll(async () => {
+  contractInstance = await setupContract();
+});
+
+describe('Auth Testing', () => {
+  it('can get is_caller_external', async () => {
+    const { value } = await contractInstance.functions.is_caller_external().call();
+
+    expect(value).toBeTruthy();
+  });
+
+  it('can check_msg_sender [with correct id]', async () => {
+    const wallet = getWallet();
+
+    const { value } = await contractInstance.functions
+      .check_msg_sender({ value: wallet.address.toB256() })
+      .call();
+
+    expect(value).toBeTruthy();
+  });
+
+  it('can check_msg_sender [with correct id, using get]', async () => {
+    const wallet = getWallet();
+    try {
+      await contractInstance.functions.check_msg_sender({ value: wallet.address.toB256() }).get();
+
+      throw new Error('it should have thrown');
+    } catch (error) {
+      expect(error).toBeTruthy();
+    }
```

Comment:
Not sure what's going on here but:
 1. Throwing inside the `try` block looks like a mistake
 2. All assertions must be outside `try/catch` blocks to ensure they always run

However, I think we could `expect(...).toThrow()` in this case:

```suggestion
    const wallet = getWallet();
    expect(() => {
      await contractInstance.functions.check_msg_sender({ value: wallet.address.toB256() }).get();
    }).toThrow();
```